Fix "API Update Required" notice due to deprecated BuildTarget.iPhone

jbroadway opened the pull request:
The "API Update Required" popup is appearing on import, and `BuildTarget.iPhone` seems to be the cause:

```
Assets/Editor/URLClient/URLClientPostprocessor.cs(38,31): error CS0619: `UnityEditor.BuildTarget.iPhone' is obsolete: `Use iOS instead (UnityUpgradable) -> iOS'
```

jbroadway committed:
Fix "API Update Required" notice due to deprecated BuildTarget.iPhone

The "API Update Required" popup is appearing on import, and `BuildTarget.iPhone` seems to be the cause:

    Assets/Editor/URLClient/URLClientPostprocessor.cs(38,31): error CS0619: `UnityEditor.BuildTarget.iPhone' is obsolete: `Use iOS instead (UnityUpgradable) -> iOS'

jbroadway committed:
Merge pull request #1 from jbroadway/patch-1

Fix "API Update Required" notice due to deprecated BuildTarget.iPhone

imkira commented:
Hi @jbroadway . Thanks for your contribution.
I think we should use an

``` csharp
#if UNITY_X_Y || ...
if (target != BuildTarget.iPhone)
#else
if (target != BuildTarget.iOS)
#endif
```

and put the list of older versions that do not support `BuildTarget.iOS`.

http://docs.unity3d.com/Manual/PlatformDependentCompilation.html

Can you figure it out what versions should it have?

jbroadway committed:
Use platform dependence to check the right build target

From what I can find, BuildTarget.iOS is Unity 5+, so I believe this should cover both cases correctly.

jbroadway committed:
Typo fix

jbroadway commented:
No problem. Just added the platform check.

imkira commented:
@jbroadway thanks a lot for the contribution.

However, I would advise you to do the reverse: to explicitly specify old versions of Unity.
The problem is, when unity 6 comes out, we will probably have to review this problem again by adding some `UNITY_6` definition or something.
I wonder if `UNITY_4` and `UNITY_3` even works though.

jbroadway commented:
Ah yes, good point. I'm travelling with only a Chromebook at the moment, so I can't test it just yet. The `UNITY_5` specifier isn't mentioned in Unity's docs on [platform dependent compilation](http://docs.unity3d.com/Manual/PlatformDependentCompilation.html), so hopefully `UNITY_3` and `UNITY_4` work too since that one does despite being undocumented. Googling didn't turn up any clear answers either, so I'll have to test once I'm back next week.

imkira commented:
No problem @jbroadway . Let me know when you have it.

jbroadway committed:
Updated platform check for future version compatibility

jbroadway commented:
Sorry that took so long! Made the change that should support future versions now :)

imkira commented:
That makes sense. Thanks!
